htdocs/mri/jiv/get_file.php

```diff
@@ -19,6 +19,46 @@
  *  @link     https://github.com/aces/Loris-Trunk
  */
 
+/**
+ * This function is to replace PHP's extremely buggy realpath().
+ * Taken from https://gist.github.com/umidjons/6154548
+ *
+ * @param string $path The original path, can be relative etc.
+ *
+ * @return string The resolved path, it might not exist.
+ */
+function truepath( $path )
+{
+    // whether $path is unix or not
+    $unipath = strlen($path) == 0 || $path{0} != '/';
+    // attempts to detect if path is relative in which case, add cwd
+    if (strpos($path, ':') === false && $unipath ) {
+        $path = getcwd() . DIRECTORY_SEPARATOR . $path;
+    }
+    // resolve path parts (single dot, double dot and double delimiters)
+    $path      = str_replace(array( '/', '\\' ), DIRECTORY_SEPARATOR, $path);
+    $parts     = array_filter(explode(DIRECTORY_SEPARATOR, $path), 'strlen');
```

Comment:
This is a pretty hard to read way to strip out the empty parts.

Can you just use a function defined inline which returns true if strlen > 0 to make it a little more obvious what the array_filter is trying to filter for, instead of basing it on an implicit assumption that 0 and false are equivalent?